core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -379,7 +379,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
    * @param hw the suggested new value for the high watermark
    * @return the updated high watermark offset
```

Comment:
We should update javadoc to doc the new return value.

Reply:
resolving as i've reverted the changes